app/ota_client.py

```diff
@@ -941,15 +944,21 @@ def ecb(e):
                 pool.close()
                 # wait for all tasks to complete
                 # not apply timeout currently
-                pool.join()
+                while len(await_c) < len(rfiles_list):
+                    # if one of the subprocess raise error,
+                    # terminate the whole pool
+                    if terminate_event.is_set():
+                        pool.terminate()
+                        break
```

Comment:
This loop is busy waiting if event is not set. I think sleep is required?

Reply:
According to the document, the **is_set** method in **Event** is non-blocking method, it will instantly return True of False depends on the event status.

Reply:
So if event is not set, this loop is busy waiting. Sleep in the loop is required I think.

Reply:
@Bodong-Yang 
This loop is just waiting the condition, so adding sleep releases the CPU resource. But I think it is optional.

Reply:
@obi-t4 We can add an interval between pulling with sleep though, it is not so time critical for this part of code. I think 1s interval is OK?